processing/src/main/java/io/druid/query/GroupByMergedQueryRunner.java

```diff
@@ -179,7 +179,7 @@ private void waitForFutureCompletion(
     try {
       queryWatcher.registerQuery(query, future);
       final long timeout = QueryContexts.getTimeout(query);
-      if (timeout == 0) {
+      if (QueryContexts.isNoTimeout(timeout)) {
```

Comment:
I think it's clearer here:
```java
if (QueryContext.hasTimeout(query)) {
  future.get(QueryContext.getTimeout(query), TimUnit.MILLISECONDS);
} else {
  future.get();
}
```

This is also true for other places where `isNoTimeout()` is used, except the one in `ChainedExecutionQueryRunner`.